### PR TITLE
chore(release): Improve release & version hygiene

### PR DESCRIPTION
**Pull request by nyergler**
This PR contains a few chore commits to improve the way we manage our dependencies and make releases.

* Updates to use pipenv, which is now the recommended way to manage virtualenvs
* Updates CI & metadata versions to support Python 3.7
* Adds bumpversion, a utility for, ahem, bumping the version consistently
* Populates the project description on https://pypi.org/project/lob/ (which is currently, sadly, empty)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
     author='Lob',
     author_email='support@lob.com',
     packages=['lob'],
     description='Lob Python Bindings',
-    url='https://www.lob.com',
+    long_description=README,
+    long_description_content_type="text/markdown",
+    url='https://github.com/lob/lob-python',
     license='MIT',
     install_requires=[
```
Comment (jhuangtw-dev) [inline setup.py]: [very optional] could this be converted to use something like:
```py
from pipenv.project import Project
from pipenv.utils import convert_deps_to_pip

pfile = Project(chdir=False).parsed_pipfile
requirements_for_setup_py = convert_deps_to_pip(pfile['packages'], r=False)
```

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,18 @@
+dist: xenial
 language: python
 python:
 - '2.7'
-- '3.4'
 - '3.5'
 - '3.6'
+- '3.7'
 - 'pypy'
```
Comment (dmlittle) [inline .travis.yml]: Should we also add `pypy3.5-6.0` since we mention in the README that we support it.

File: lob/version.py [hunk 3]
```diff
@@ -1,3 +1,3 @@
 from __future__ import unicode_literals
 
-VERSION = '3.4.0'
+VERSION = '4.0.0'
```
Comment (dmlittle) [inline lob/version.py]: It doesn't look like we're changing any behavior to the library, is there a specific reason why we need to bump the version?

Comment (nyergler) [reply lob/version.py]: `4.0.0` was released back in October, but the code wasn't updated. Whomp whomp. This PR just brings us into sync with reality.

File: .travis.yml (hunk 2 above)
Comment (nyergler) [reply .travis.yml]: Added `pypy3.5` to the list; not sure what `6.0` means here.

**Pull request merged**